sdk/iothub/src/service/twin.rs

```diff
@@ -0,0 +1,419 @@
+use azure_core::errors::{extract_status_headers_and_body, AzureError, UnexpectedHTTPResult};
+use hyper::{Body, Client, Method, Request, StatusCode};
+use hyper_tls::HttpsConnector;
+use serde::de::{self};
+use serde::{Deserialize, Deserializer};
+use std::collections::HashMap;
+use std::marker::PhantomData;
+
+use crate::service::{ServiceClient, API_VERSION};
+
+/// AuthenticationType of a module or device
+#[derive(Debug)]
+pub enum AuthenticationType {
+    Certificate,
+    Authority,
+    None,
+    SAS,
+    SelfSigned,
+}
+
+impl<'de> Deserialize<'de> for AuthenticationType {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        match s.as_str() {
+            "certificate" => Ok(AuthenticationType::Certificate),
+            "sas" => Ok(AuthenticationType::SAS),
+            "Authority" => Ok(AuthenticationType::Authority),
+            "selfSigned" => Ok(AuthenticationType::SelfSigned),
+            "none" => Ok(AuthenticationType::None),
+            _ => Err(de::Error::custom(format!("Expected status to be 'certificate','sas','Authority','selfSigned' or 'none' but received: {}", s))),
+        }
+    }
+}
+
+/// The connection state of a module or device
+#[derive(Debug)]
+pub enum ConnectionState {
+    Connected,
+    Disconnected,
+}
+
+impl std::fmt::Display for ConnectionState {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            ConnectionState::Connected => write!(f, "connected"),
+            ConnectionState::Disconnected => write!(f, "disconnected"),
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for ConnectionState {
```

Comment:
Why not `#[derive(Deserialize)]` for this type?

Reply:
To be honest I thought it wasn't possible for `enum` types in the way I wanted when I looked at the docs back when I was implementing this. It does work though so I'll get this changed asap.